codestyle/guava-forbidden-apis.txt

```diff
@@ -0,0 +1 @@
+com.google.common.util.concurrent.Futures.transform(com.google.common.util.concurrent.ListenableFuture<I>, com.google.common.util.concurrent.AsyncFunction<? super I,? extends O>) @ Use io.druid.java.util.common.concurrent.ListenableFutures.transform
```

Comment:
I don't think it needs to be a different file, `druid-forbidden-apis.txt` already forbids Guava APIs.

Reply:
Forbid `Iterators.emptyIterator()`

Reply:
And still - 

> I don't actually see Futures.transform() is gone. It seems to me that it's deprecated, but not gone.

And won't be removed, if Guava keeps the current policy of non-removing APIs.

Reply:
The async one is gone, or at least changed to a new method name, I think `asyncTransform`. The method for the non-async one is still there and unchanged.

Reply:
I see now. Could you please

 - Match the method name in ListenableFutures with the present name in Guava library.
 - Add comments to the method, and to the entry in forbidden-apis file (it supports comments with `#`), that they should be removed, with switch back to Guava, as soon as the minimum supported Guava version becomes 19.0.

Reply:
That sounds good

Reply:
fixed